addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/api/model/UniFiSite.java

```diff
@@ -23,6 +23,12 @@
  */
 public class UniFiSite {
 
+    private final transient UniFiController controller;
+
+    public UniFiSite(UniFiController controller) {
+        this.controller = controller;
```

Comment:
The `controller` field doesn't seem to be used here. Is it something for part of future support? Same question for other classes

Reply:
> Is it something for part of future support?

Exactly! The model classes have access to the controller directly which results in a cleaner API:

```java
public class UniFiSiteThingHandler {
  public void someMethod() {
    UniFiSite site = ...
    site.getTotalClients();
  }
}

public class UniFiSite {
  public long getTotalClients() {
    return controller.getTotalClients(this);
  }
}

public class UniFiController {
  public long getTotalClients(UniFiSite site) {
    ...
  }
}
```

This means all the `UniFiXXXThingHandlers` do not need direct access to the controller, just the entitles they are "managing". The controller access is encapsulated in the model classes.